### DISCUSSION TITLE
[F.R] Add a achievement badge for sponsorship

### DISCUSSION THREAD

**Discussion by huabin** [community/community#18313] [Other Feature Feedback, Questions, & Ideas]
Hello,

I found the new profile, it's cool and good job.

A few months ago, I asked your customer service why there’s no sponsorship information that I had sponsored on my profile, and he told me that it was because I was a one-time sponsor, not an ongoing sponsor.

Seeing the new badges, I wonder if it was possible to add a badge for sponsorship like following.

![eg](https://user-images.githubusercontent.com/535675/173165431-6f9655c1-44d1-4dd9-bd43-dec27da0f8b4.jpg)

Hope this will be a good suggestion. Thank you.

**Comment by Nishant1500** [comment]
If i remember correctly, there is already a sponsorship badge. Have a look at this -> https://github.com/Schweinepriester/github-profile-achievements#achievements

**Reply by huabin** [reply]
Yes, but there's an issue. If you are a one-time sponsor, you will lose the badge after a few days.

**Reply by Nishant1500** [reply]
Oh.

**Comment by nel0x** [comment]
Hi @huabin 

Unfortunately I think that was the intention in order to motivate folks to setup long-term subscriptions which are reliable for the developers ...
Also on another focus: Are you sure that the "Public Sponsor" Achievement disappears on one-time sponsorings?
On another account once I did at least one or two one-time sponsorings but the badge certanly didn't go away

Hope this helps!

**Comment by huabin** [comment]
Hello @nel0x

In my opinion, sponsorships is not a game. Anyone who has sponsored others
should be rewarded.

I'm sure the one-time sponsorship badge will disappear after a while, I
checked this with GitHub support agent.

---
Bin Hua

**Reply by nel0x** [reply]
Should they? Would mean everyone could do 1$ one-time and have the badge ...
Understand you, just want to say why I think it's on purpose

**Reply by huabin** [reply]
Of course, they should.

Who would achieve their purpose by sponsoring $1 to a developer?
BTW, what is the purpose of a one-time sponsor? 
Is it to show off that they are rich enough to sponsor some developer $1?

If badge is only for long term sponsors. Then at least show on the profile of the one-time sponsor that they have sponsored certain projects. 
Such as here
![d](https://user-images.githubusercontent.com/535675/175809573-df547c52-932c-4691-9c9d-cd7a963df83b.png)

They deserve it.

**Comment by huabin** [comment]
BTW, on the profile page, there isn't even a place to show that I have sponsored someone. 

That's kind of wrong.

**Comment by rilysh** [comment]
That's not how sponsorship works. If you stop supporting someone/an organisation, that would make sense why GitHub will remove the sponsorship badge from your profile. Just support them, in monthly budget. Obviously GitHub will not do anything like that.

> BTW, on the profile page, there isn't even a place to show that I have sponsored someone.

If you've sponsored someone they'll know that. There's no reason to log the past sponsorship. Meanwhile, it will increase one time sponsorship which will not help an open source developer. And that's the main reason why it's not like other badges that we currently have.

Hope this clarifies.

**Comment by huabin** [comment]
@Ruzie 

What makes you think that a one-time sponsorship won't help developers?

**Reply by rilysh** [reply]
> @Ruzie
> 
> What makes you think that a one-time sponsorship won't help developers?

How about if I donate only *$5* for a open source project and never do any sponsorship with that person?
It obviously not going to help, correct? That what I wrote about.

**Comment by nel0x** [comment]
@huabin we just have different opinions about that. You have to understand the following:
Of course a 1000$ one-time sponsorship will help the developert, even 100 bugs or 50 will. But most people won't pay so much (at once). And implementing different stages and minimum amounts will just get complicated and messed up.

What helps developers most are *undoubtably reliable subscriptions*, even with small amounts. That's what Github (as well as other platforms) try to push and motivate its users for. Because that's the most reliable and sustainable option, and even for the sponsors not very budget-heavy (at least short term ...). Therefore that is just the *best* - but at all means not *perfect* - solution.

**Comment by huabin** [comment]
@nel0x Your reply is beyond the topic of our discussion.

We are discussing whether there should be a way to display a badge or something similar on a one-time sponsor profile page that indicates that the person has sponsored others before.

In my opinion, each sponsorship should be commended, whether one-time or long-term. This is what will help open source developers and the open source community.

**Comment by huabin** [comment]
@nel0x 

To give you an example, I have sponsored curl once, which I consider to be an honor to me.

But I found that it doesn't show up on my profile, or even on the curl's sponsorship page. What's worse is that former sponsor who have deleted her account show up on it, but I don't, which is so weird.

![error?](https://user-images.githubusercontent.com/535675/175811193-d749c917-897c-4d98-9a3b-af968498773f.png)

![error2](https://user-images.githubusercontent.com/535675/175811219-619046c5-b0f8-4ee9-84a7-17f7ca2c746e.png)

**Comment by nel0x** [comment]
Are you sure you did the sponsoring public and didn't select private?

Did you have the badge even for a few hours/days and was visible as a sponsor for that time? If not that wasn't a public payment.

**Comment by huabin** [comment]
@nel0x 

Of course, I used to have the badge for a few days and then I lost it. My avatar was also displayed on the curl's sponsorship page, and then it disappeared as well.

I checked with GitHub support agent, and I posted this discussion to try to provide a solution.

**Comment by rilysh** [comment]
@huabin Just in case, if you're thinking about you'll sponsor only one time and the sponsorship badge (to show up in public) will stay forever, that's not. That will be like if I give you $10 one day and you'll need to proud forever.

This will grow up and GitHub badge hunters will do one time sponsorship for the badge.

**Comment by huabin** [comment]
@Ruzie

Hunting badges for what?

In your opinion, $5 is not a help to an open source developer, is it?

Even if it's just one dollar, it's a help to the developers and the
community.

On Sun, Jun 26, 2022 at 19:26 Ruzie ***@***.***> wrote:

> @huabin <https://github.com/huabin> Just in case, if you're thinking
> about you'll sponsor only one time and the sponsorship badge (to show up in
> public) will stay forever, that's not. That will be like if I give you $10
> one day and you'll need to proud forever.
>
> This will grow up and GitHub badge hunters
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/github-community/discussions/18313#discussioncomment-3025739>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAECY63ZNQQKPHWNYBVHXELVRA47VANCNFSM5YPGMZYQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> .com>
>
-- 

Sent from my iPad

**Comment by huabin** [comment]
@Ruzie

Do you think people who sponsor more than $500 through a one-time
sponsorship deserve a badge?

On Sun, Jun 26, 2022 at 19:53 Bin Hua ***@***.***> wrote:

> @Ruzie
>
> Hunting badge for what?
>
> In your opinion, $5 is not a help to an open source developer, is it?
>
> Even if it's just one dollar, it's a help to the developers and the
> community.
>
> On Sun, Jun 26, 2022 at 19:26 Ruzie ***@***.***> wrote:
>
> > @huabin <https://github.com/huabin> Just in case, if you're thinking
> > about you'll sponsor only one time and the sponsorship badge (to show up
> in
> > public) will stay forever, that's not. That will be like if I give you
> $10
> > one day and you'll need to proud forever.
> >
> > This will grow up and GitHub badge hunters
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/orgs/github-community/discussions/18313#discussioncomment-3025739
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AAECY63ZNQQKPHWNYBVHXELVRA47VANCNFSM5YPGMZYQ
> >
> > .
> > You are receiving this because you were mentioned.Message ID:
> > ***@***.***
> > .com>
> >
> --
>
> Sent from my iPad
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/github-community/discussions/18313#discussioncomment-3025810>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAECY6YNYP3DUH7YISHWIMTVRBAD5ANCNFSM5YPGMZYQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: <github-community/community/repo-discussions/18313/comments/3025810@
> github.com>
>
-- 

Sent from my iPad

**Reply by rilysh** [reply]
> @Ruzie Do you think people who sponsor more than $500 through a one-time sponsorship deserve a badge?
> On Sun, Jun 26, 2022 at 19:53 Bin Hua ***@***.***> wrote: @Ruzie Hunting badge for what? In your opinion, $5 is not a help to an open source developer, is it? Even if it's just one dollar, it's a help to the developers and the community. On Sun, Jun 26, 2022 at 19:26 Ruzie ***@***.***> wrote: > @huabin <https://github.com/huabin> Just in case, if you're thinking > about you'll sponsor only one time and the sponsorship badge (to show up in > public) will stay forever, that's not. That will be like if I give you $10 > one day and you'll need to proud forever. > > This will grow up and GitHub badge hunters > > — > Reply to this email directly, view it on GitHub > < https://github.com/orgs/github-community/discussions/18313#discussioncomment-3025739 >, > or unsubscribe > < https://github.com/notifications/unsubscribe-auth/AAECY63ZNQQKPHWNYBVHXELVRA47VANCNFSM5YPGMZYQ > > . > You are receiving this because you were mentioned.Message ID: > ***@***.*** > .com> > -- Sent from my iPad — Reply to this email directly, view it on GitHub <https://github.com/orgs/github-community/discussions/18313#discussioncomment-3025810>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAECY6YNYP3DUH7YISHWIMTVRBAD5ANCNFSM5YPGMZYQ> . You are receiving this because you are subscribed to this thread.Message ID: <github-community/community/repo-discussions/18313/comments/3025810@ github.com>
> --
> […](#)
> Sent from my iPad

At first you don't need to mention two times, thanks.

Second, no, in my opinion, if you stop sponsoring you'll no longer have that badge, as most of people won't just give $500.

Badge hunters, clarifies itself what is that, I don't need to mention.

**Comment by huabin** [comment]
Again, hunting badge for what? Make it clear, I really don’t get the point
of what you’re saying.

Think about this, $500 = one-time amount * times * projects.

Not big deal.

On Sun, Jun 26, 2022 at 20:05 Ruzie ***@***.***> wrote:

> @Ruzie <https://github.com/Ruzie> Do you think people who sponsor more
> than $500 through a one-time sponsorship deserve a badge?
> On Sun, Jun 26, 2022 at 19:53 Bin Hua *@*.*> wrote: @Ruzie
> <https://github.com/Ruzie> Hunting badge for what? In your opinion, $5 is
> not a help to an open source developer, is it? Even if it's just one
> dollar, it's a help to the developers and the community. On Sun, Jun 26,
> 2022 at 19:26 Ruzie @.*> wrote: > @huabin <https://github.com/huabin>
> https://github.com/huabin Just in case, if you're thinking > about you'll
> sponsor only one time and the sponsorship badge (to show up in > public)
> will stay forever, that's not. That will be like if I give you $10 > one
> day and you'll need to proud forever. > > This will grow up and GitHub
> badge hunters > > — > Reply to this email directly, view it on GitHub > <
> https://github.com/orgs/github-community/discussions/18313#discussioncomment-3025739
> >, > or unsubscribe > <
> https://github.com/notifications/unsubscribe-auth/AAECY63ZNQQKPHWNYBVHXELVRA47VANCNFSM5YPGMZYQ
> > > . > You are receiving this because you were mentioned.Message ID: >
> *@*.*** > .com> > -- Sent from my iPad — Reply to this email directly,
> view it on GitHub
> https://github.com/orgs/github-community/discussions/18313#discussioncomment-3025810,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/AAECY6YNYP3DUH7YISHWIMTVRBAD5ANCNFSM5YPGMZYQ
> . You are receiving this because you are subscribed to this thread.Message
> ID: <github-community/community/repo-discussions/18313/comments/3025810@
> github.com>
>
> … <#m_-4457342506867286940_>
> Sent from my iPad
>
> At first you don't need to mention two times, thanks.
>
> Second, no, in my opinion, if you stop sponsoring you'll no longer have
> that badge, as most of people won't just give $500.
>
> Badge hunters, clarifies itself what is that, I don't need to mention.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/github-community/discussions/18313#discussioncomment-3025842>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAECY6ZDRURPVBQRERTJ443VRBBSDANCNFSM5YPGMZYQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> .com>
>
-- 

Sent from my iPad

**Comment by nel0x** [comment]
It's just that most people will NOT donate 500$.
More like 2 - 20$
And for those amounts it is great to do one-time donations but don't expect to get a (permanent) "public sponsor" badge. Thats only for those being ACTIVE thus regular sponsors.

As written before you could say > 50 one-time is the same bla bla... but that would make things just messed up, so tldr:
When you do one-time you should get a badge for one/few weeks IMHO but obly While active sponsoring (not months ago) you'll have the badge.
Why the deleted axcount there is in the sponsor list there seems strange to me too.

**Comment by buddydeus** [comment]
I have a question, whether it's $1 or $1,000, isn't it my payment?
I paid, can't I get back a virtual badge? Does that mean that a coder who only contributed 1 line of code in the future cannot be called a code contributor?

So, getting the (permanent) "public sponsor" is really unreasonable, but getting other badge to prove that I have paid is not too much

Because it will get more people involved, even badge hunters. This is in support of the project

**Reply by huabin** [reply]
Good point, does a programmer who contributed a single line of code qualify as a contributor to this project?

**Reply by nel0x** [reply]
Generally code can't be measured so easily. One line of code could be *the* important patch to an extremely dangerous exploit while 100 lines can be BS ...
But that's right, code quality can also be subjective

**Comment by nel0x** [comment]
> Does that mean that a coder who only contributed 1 line of code in the future cannot be called a code contributor?

Exactly! You've got it!
Of course your a "code contributor" when you contribute one character of code, and of course your technically a "sponsor" when you donate 1 cent.
But when do we want to reward that!? Should every click on Github be rewarded with a "Github active member" badge!?
No, there needs to be a barrier of entry. Not too high so that people get frustrated and it's out of reach but high enough to be a challange you can be proud of.
And the barrier the GitHub team settled on for the sponsoring badge is: must be a *active* (=> regular) sponsor, because in the very most cases that will benefit developers the most.

> Because it will get more people involved, even badge hunters. This is in support of the project

With badges for one-time donations most members (including badge hunters) will support a project one time, preferably with the minimal amount required and after the first month projects will barely receive anything anymore.

**Comment by buddydeus** [comment]
Why? Why do you need a threshold?

The point of the badge is to inspire more people to get involved and support the program!

How many people use github? 0.01% of people donate $1, which is all very objective money for the project.
But because I only donated $1, my badge is gone after 1 month? How else can I motivate more people? Including how to motivate me to donate a second, nth time?
I might forget about the project 1 month later, but the badge will remind me of it and I will donate again!

So a badge is a proof and a hope, don't set any threshold for hope!

**Reply by sanyam-katoch10** [reply]
Badge is temporary unless you renew payment every month

**Comment by huabin** [comment]
Just found that this suggestion was accepted, and all one-time sponsors won.

This discussion can be locked now. Cool.

**Comment by Nishant1500** [comment]
Yey finally :)))

**Comment by sanyam-katoch10** [comment]
Actually new bagde is added for sponsorship achievement 
Very cool
